packages/docbook_xml42.rb

```diff
@@ -0,0 +1,32 @@
+require 'package'
+
+class Docbook_xml42 < Package
+  description 'Version 4.2 - document type definitions for verification of XML data files against the DocBook rule set'
+  compatibility 'all'
+  homepage 'http://www.docbook.org'
+  version '4.2'
+  source_url 'http://www.oasis-open.org/docbook/xml/4.2/docbook-xml-4.2.zip'
+  source_sha256 'acc4601e4f97a196076b7e64b368d9248b07c7abf26b34a02cca40eeebe60fa2'
+
+  depends_on 'docbook'
+  depends_on 'xmlcatmgr'
+  depends_on 'docbook_xml'
+  depends_on 'docbook_xsl' # Requires the catalog.xml created within this package
+  
+  def self.preinstall
+    system "sed -i -e 's,<!-- .* -->,,g' #{CREW_PREFIX}/etc/xml/catalog.xml"
+    system "cat << EOF > ./remove_add.sh
+xmlcatmgr -c /usr/local/etc/xml/docbook remove public '-//OASIS//DTD DocBook XML V4.2//EN' '/usr/local/share/xml/docbook/4.2/catalog.xml'
+xmlcatmgr -c /usr/local/etc/xml/docbook remove public '-//OASIS//DTD DocBook XML V4.1.2//EN' '/usr/local/share/xml/docbook/4.2/catalog.xml'
+xmlcatmgr -c /usr/local/etc/xml/docbook add public '-//OASIS//DTD DocBook XML V4.2//EN' '/usr/local/share/xml/docbook/4.2/catalog.xml'
+xmlcatmgr -c /usr/local/etc/xml/docbook add public '-//OASIS//DTD DocBook XML V4.1.2//EN' '/usr/local/share/xml/docbook/4.2/catalog.xml'
+
+EOF"
+    system "bash ./remove_add.sh"
```

Comment:
I'm not following.  Why do we need to remove and then add?  And why does it need to be a script?

Reply:
We're not installing a file here, but editing on that's already been installed. The issue is, upon removal, the edited line is still there, and will cause compilation to fail upon reinstallation. In #4315 I mentioned how a simple fix would be to use a bash script, that's what I've done here.

Reply:
Removing the line can't be done if it's not present, which would cause compilation to fail.
Please read from here to the end.
> https://github.com/skycocker/chromebrew/issues/4315#issuecomment-691685691

Reply:
You could run the commands explicitly without a script, correct?  Since this is in the prebuild section, it is only run during the build.  Don't you want this to run during install/reinstall?  If that is the case, then these commands should be in the preinstall section instead.

Reply:
if they were to both be placed in postinstall, after everything has been installed, it would work without a script.

Reply:
This is creating a one-time usage script, thas only needed because the removal command can sometimes fail. The command edits a file installed by `docbook_xml51`, it really doesn't matter which section we put them in.

Reply:
OK, so prebuild is the correct location then.